### ISSUE TITLE
Stopping & restarting the sequencer confuses stream reads & writes

### ISSUE THREAD

**Issue by slfritchie** [CorfuDB/CorfuDB#206]
Based on `master` branch at commit 8383e89b9bbf22be839b09ce6f85db58eaa622ec:
1. Write data to a stream
2. Restart server
3. Read stream.  It's empty. (sadpanda)
4. Write to stream.
5. Read stream.  It has only data from step 4. (sadpanda)

A self-contained shell script to demonstrate this behavior is at https://gist.github.com/slfritchie/10b02ee8a86b12f347cf7597790673f2

Output of the script, minus the `INFO` log messages.

```
Start server
Write hellos
Stop and restart server
We expect 3 hello world lines below:
Write more data
We expect 6 lines total below
More data at Wed Jul 27 18:18:05 PDT 2016
More data at Wed Jul 27 18:18:07 PDT 2016
More data at Wed Jul 27 18:18:09 PDT 2016
```

**Comment by slfritchie** [comment]
November review: keep open.

**Comment by slfritchie** [comment]
Mid-June issues review: fixed by #558 and related PRs.
